activestorage/app/models/active_storage/blob.rb

```diff
@@ -21,6 +21,7 @@ class ActiveStorage::Blob < ActiveStorage::Record
 
   has_secure_token :key, length: MINIMUM_TOKEN_LENGTH
   store :metadata, accessors: [ :analyzed, :identified, :composed ], coder: ActiveRecord::Coders::JSON
+  serialize :checksum, coder: ActiveStorage::Checksum
```

Comment:
One issue with this is that after upgrading all existing blob will be marked as `changed?`. In general it's not a huge deal, but it can cause some nasty corner cases.

It could be interesting to ensure `dump/load` perfectly rountrip old values.

Reply:
If it is important to preserve the raw MD5 value in the DB moving forward I can add a case to `dump` to write out in that format.